open-metadata-implementation/access-services/asset-catalog/asset-catalog-client/src/main/java/org/odpi/openmetadata/accessservices/assetcatalog/AssetCatalog.java

```diff
@@ -172,6 +174,34 @@ public AssetListResponse searchByType(String userId,
         return assetResponse;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AssetListResponse searchByTypeNameOrGUID(String userId,
```

Comment:
I do not think is great to have a method 'something1 or something2' then working out which one it is is by a null parameter.  I think it would be much cleaner to have separate methods one for get by type name and one for get by type guid. and call common code in the methods.